### PR TITLE
4.0 - Add select to endpoints that use process_array 

### LINKED ISSUE #5075: Add select to ruleset endpoints for new API
Hello team,

We should study adding select parameter to some of our ruleset related endpoints. These endpoints use framework function process_array to format and filter the response, we should add select option to process array and the controllers for some of the endpoints.

Review adding select to the following endpoints:

- [x] `/rules`
- [x] `/decoders`
- [x] `/lists`

### PR DESCRIPTION
**Pull request by vicferpoy**
Hello team, this closes #5075 .

This PR adds the function `select_array` to the `utils.py` module, making possible the use of the parameter `select` on endpoints that use `process_array` instead of `WazuhDBQuery`.

In addition to this, I have added integration tests for every modified endpoint and the unit tests for `utils.py` (This module needs new tests since they are outdated). These are the endpoints that include the paramater `select` now:
- `GET /rules`
- `GET /decoders`
  `GET /decoders/parents`
- `GET /lists`

In order to run the integration tests, this PR must be merged before: https://github.com/wazuh/wazuh/pull/5076

I used the test functions this PR includes to verify my responses.

# Tests performed
## Integration tests
```
Collected tests [3]:
test_decoders_endpoints.tavern.yaml, test_lists_endpoints.tavern.yaml, test_rules_endpoints.tavern.yaml


test_decoders_endpoints.tavern.yaml 
	 23 passed, 29 warnings

test_lists_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rules_endpoints.tavern.yaml 
	 11 passed, 13 warnings

``` 

## Unit tests
```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.2, pytest-5.4.2, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/vicferpoy/Desktop/Git/wazuh/framework
plugins: tavern-1.0.0
collected 3 items                                                                                                                                                                                                 

test_utils.py::test_select_array[select0-required_fields0-expected_result0] PASSED                                                                                                                          [ 33%]
test_utils.py::test_select_array[select1-None-None] PASSED                                                                                                                                                  [ 66%]
test_utils.py::test_select_array[select2-None-None] PASSED                                                                                                                                                  [100%]

================================================================================================ 3 passed in 0.08s ================================================================================================

```

Regards.

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 72aa764dd3978da5e7c8f02d5aca07a3d5c87154
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2152/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2152/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2152/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2152.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2152/Analysis_dynamic_valgrind-v4.0.0-B2152.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 5ca644afbddd0ab5d68da7f7695fff0985c1bb53
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2198/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2198/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2198/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2198.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2198/Analysis_dynamic_valgrind-v4.0.0-B2198.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 9e6a7f19a370268c079ba8887f0b19169388484a
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2205/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2205/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2205/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2205.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2205/Analysis_dynamic_valgrind-v4.0.0-B2205.log)

File: api/api/controllers/decoders_controller.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ async def get_decoders(request, decoder_names: list = None, pretty: bool = False
     :param wait_for_complete: Disable timeout response
     :param offset: First element to return in the collection
     :param limit: Maximum number of elements to return
+    :param select: Select which fields to return (separated by comma)
```
Comment (davidjiglesias) [inline api/api/controllers/decoders_controller.py:31]:
```suggestion
    :param select: List of selected fields to return
```

File: framework/wazuh/cdb_list.py [hunk 2]
```diff
@@ -41,7 +43,8 @@ def get_lists(path=None, offset=0, limit=common.database_limit, sort_by=None, so
 
     data = process_array(lists, search_text=search_text, search_in_fields=search_in_fields,
                          complementary_search=complementary_search, sort_by=sort_by, sort_ascending=sort_ascending,
-                         offset=offset, limit=limit, allowed_sort_fields=['relative_dirname', 'filename'])
+                         offset=offset, limit=limit, select=select, allowed_sort_fields=['relative_dirname', 'filename'],
```
Comment (davidjiglesias) [inline framework/wazuh/cdb_list.py:46]: Add SORT_FIELDS to core/cdb_list and use it here for allowed_sort_fields

File: framework/wazuh/core/cdb_list.py [hunk 3]
```diff
@@ -9,6 +9,8 @@
 from wazuh import common
 from wazuh.exception import WazuhError
 
+REQUIRED_FIELDS = ['relative_dirname', 'filename']
+
```
Comment (davidjiglesias) [inline framework/wazuh/core/cdb_list.py:13]: Add SORT_FIELDS here

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 2fbd0ea437b5dc7f1360b3caf6bb33dac4260b57
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2260/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2260/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2260/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2260.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2260/Analysis_dynamic_valgrind-v4.0.0-B2260.log)

**Pull request merged**